makedmdconf.d

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env rdmd
+module makedmdconf;
+
+import core.stdc.stdlib : exit;
+import std.getopt, std.string;
+import std.path, std.file, std.stdio;
+import std.process;
+
+int main(string[] args)
```

Comment:
I'd prefer to have this file integrated into `build.d` as a new target.  The logic for determining whether or not the target needs to be generated and the OS detection logic is already there, and I don't see much of a use case for generating the `dmd.conf` file outside of a typical build.  The makefile can also generate the `dmd.conf` file by simply calling `build.d conf` (or whatever the target is called).